lib/assignment/widgets/assignments_screen.dart

```diff
@@ -153,39 +160,53 @@ class AssignmentCard extends StatelessWidget {
       children: <Widget>[
         Expanded(
           child: FancyText(
-            assignment.name,
+            widget.assignment.name,
             style: context.textTheme.subhead,
             maxLines: 2,
           ),
         ),
-        if (assignment.dueAt != null) ...[
+        if (widget.assignment.dueAt != null) ...[
           SizedBox(width: 8),
           Text(
-            assignment.dueAt.shortDateTimeString,
+            widget.assignment.dueAt.shortDateTimeString,
             style: context.textTheme.caption,
           ),
         ],
       ],
     );
   }
 
+  Widget _buildCourseChip(BuildContext context) {
+    if (_course != null) {
+      return CourseChip(
+        _course,
+        onPressed: () {
+          // TODO(JonasWanke): filter list by course, https://github.com/schul-cloud/schulcloud-flutter/issues/145
+        },
+      );
+    }
+    return widget.assignment.courseId != null
+        ? EntityBuilder<Course>(
+            id: widget.assignment.courseId,
+            builder: handleError((context, course, fetch) {
+              _course ??= course;
+              return CourseChip(
+                course,
+                onPressed: () {
+                  // TODO(JonasWanke): filter list by course, https://github.com/schul-cloud/schulcloud-flutter/issues/145
+                },
+              );
+            }),
+          )
+        : null;
+  }
```

Comment:
Maybe put this method below the actual `build` method? In most other code, our structure of helper-build methods is top-down.